### DISCUSSION TITLE
[@types/node] `net.Socket.localAddress` is wrongly optional

### DISCUSSION THREAD

**Discussion by bpasero** [DefinitelyTyped/DefinitelyTyped#57246] [Issues with a @types package]
It looks like the `@types/node: 16.x` library wrongly sets `net.Socket.localAddress` to optional, but I see no mention of that in the official documentation:

https://nodejs.org/api/net.html#socketlocaladdress

[a comment by an automated account is not shown]

**Comment by SimonSchick** [comment]
The documentation is.. not always accurate unfortunately, rest assured this value is not always set, feel free to check the [source](https://github.com/nodejs/node/blob/f57010909435d6bfa63398215575fcb185269c14/lib/net.js#L755).

**Reply by bpasero** [reply]
Oh ok, I was not aware the docs could be wrong...

**Reply by LinusU** [reply]
I don't think that the Node.js documentation ever specifies nullability/undefinedabilty?

**Reply by bpasero** [reply]
It does, see: https://nodejs.org/docs/v17.0.0/api/child_process.html#subprocesspid

**Reply by LinusU** [reply]
Ah, that's cool! You could submit a PR to the Node.js docs to update `localAddress` 🙌
